### PR TITLE
[WIP] Ip reconciler alerting secure metrics

### PR DESCRIPTION
**Pull request by nicklesimba**
Attempting to enable secure metrics for whereabouts by replacing http with https. Do not merge.

### CODE REVIEW

File: bindata/network/multus/multus.yaml [hunk 1]
```diff
@@ -521,49 +524,63 @@ spec:
...
+            - name: linuxptp-certs
+              mountPath: /etc/metrics
+              readOnly: True
+      priorityClassName: openshift-user-critical
+      serviceAccountName: multus
+      volumes:
+        - name: system-cni-dir
+          hostPath:
+            path: {{ .SystemCNIConfDir }}/whereabouts.d
+        - name: linuxptp-certs
```
Comment (aneeshkp) [inline bindata/network/multus/multus.yaml:583]: use `whereabouts-secret`

Comment (aneeshkp) [inline bindata/network/multus/multus.yaml:583]: name it as `whereabouts-cert`

File: bindata/network/multus/whereabouts-monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+kind: Service
+metadata:
+  name: whereabouts
+  namespace: openshift-multus
+  labels:
+    app: whereabouts
+  annotations:
+    service.alpha.openshift.io/serving-cert-secret-name: whereabouts-secret
+    prometheus.io/scrape: "true"
+    service.beta.openshift.io/serving-cert-secret-name: linuxptp-daemon-secret
```
Comment (aneeshkp) [inline bindata/network/multus/whereabouts-monitor.yaml:38]: remove this line

File: bindata/network/multus/multus.yaml [hunk 3]
```diff
@@ -521,49 +524,63 @@ spec:
...
+            - --logtostderr
+            - --secure-listen-address=:8443
+            - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
+            - --upstream=http://127.0.0.1:1984/
+            - --tls-private-key-file=/etc/metrics/tls.k
+          ports:
+            - containerPort: 8443
+              name: https
+          volumeMounts:
+            - name: linuxptp-certs
```
Comment (aneeshkp) [inline bindata/network/multus/multus.yaml:574]: name is as `whereabouts-cert`

File: bindata/network/multus/multus.yaml [hunk 4]
```diff
@@ -521,49 +524,63 @@ spec:
...
+          - name: system-cni-dir
+            mountPath: /host/etc/cni/net.d/whereabouts.d/
+        - name: kube-rbac-proxy
+          image: quay.io/coreos/kube-rbac-proxy:v0.5.0
+          args:
+            - --logtostderr
+            - --secure-listen-address=:8443
+            - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
+            - --upstream=http://127.0.0.1:1984/
+            - --tls-private-key-file=/etc/metrics/tls.k
```
Comment (aneeshkp) [inline bindata/network/multus/multus.yaml:569]: typo .key is missing 
- --tls-private-key-file=/etc/metrics/tls.key

File: bindata/network/multus/whereabouts-monitor.yaml (hunk 2 above)
Comment (nicklesimba) [reply bindata/network/multus/whereabouts-monitor.yaml:38]: Which line did you want removed here? And does it matter whether the annotation field name uses "alpha" or "beta"?

File: bindata/network/multus/multus.yaml (hunk 4 above)
Comment (nicklesimba) [reply bindata/network/multus/multus.yaml:569]: Do I need the following line as well?

```- --tls-cert-file=/etc/metrics/tls.crt```

File: bindata/network/multus/whereabouts-monitor.yaml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  name: prometheus-k8s
+  namespace: openshift-multus
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: prometheus-k8s
+subjects:
+- kind: ServiceAccount
+  name: prometheus-k8s
+  namespace: openshift-monitoring
```
Comment (aneeshkp) [inline bindata/network/multus/whereabouts-monitor.yaml:83]: add new line

**Comment by nicklesimba** [comment]
Closing this one out so I don't accidentally use it again...

**Pull request closed without merging**